browser/extensions/api/brave_rewards_api.cc

```diff
@@ -270,5 +270,107 @@ void BraveRewardsGetACEnabledFunction::OnGetACEnabled(bool enabled) {
+  RewardsService* rewards_service_ =
+    RewardsServiceFactory::GetForProfile(profile);
+
+  if (rewards_service_) {
+    rewards_service_->SaveRecurringDonation(
+      params->publisher_key, params->new_amount);
+  }
+
+  return RespondNow(NoArguments());
+}
+
+BraveRewardsRemoveRecurringDonationFunction::
+~BraveRewardsRemoveRecurringDonationFunction() {
+}
+
+ExtensionFunction::ResponseAction
+  BraveRewardsRemoveRecurringDonationFunction::Run() {
+
+  std::unique_ptr<brave_rewards::RemoveRecurringDonation::Params> params(
+    brave_rewards::RemoveRecurringDonation::Params::Create(*args_));
+
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  RewardsService* rewards_service_ =
+    RewardsServiceFactory::GetForProfile(profile);
+
+  if (rewards_service_) {
+    rewards_service_->RemoveRecurring(params->publisher_key);
+  }
+
+  return RespondNow(NoArguments());
+}
+
+BraveRewardsGetRecurringDonationsFunction::
+~BraveRewardsGetRecurringDonationsFunction() {
+}
+
+ExtensionFunction::ResponseAction
+BraveRewardsGetRecurringDonationsFunction::Run() {
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  RewardsService* rewards_service =
+    RewardsServiceFactory::GetForProfile(profile);
+
+  if (!rewards_service) {
+    return RespondNow(Error("Rewards service is not initialized"));
+  }
+
+  rewards_service->GetRecurringDonationsList(base::Bind(
+        &BraveRewardsGetRecurringDonationsFunction::OnGetRecurringDonations,
+        this));
+  return RespondLater();
+}
+
+void BraveRewardsGetRecurringDonationsFunction::OnGetRecurringDonations(
+    std::unique_ptr<::brave_rewards::ContentSiteList> list) {
+  std::unique_ptr<base::DictionaryValue> result(new base::DictionaryValue());
+  auto recurringDonations = std::make_unique<base::ListValue>();
+
+  if (!list->empty()) {
+    for (auto const& item: *list) {
+      auto recurringDonation = std::make_unique<base::DictionaryValue>();
+      recurringDonation->SetString("publisherKey", item.id);
+      recurringDonation->SetInteger("amount", item.weight);
+      recurringDonations->Append(std::move(recurringDonation));
+    }
+  }
+
+  result->SetList("recurringDonations", std::move(recurringDonations));
+  Respond(OneArgument(std::move(result)));
+}
+
+BraveRewardsGetPublisherDonationAmountsFunction::
+~BraveRewardsGetPublisherDonationAmountsFunction() {
+}
+
+ExtensionFunction::ResponseAction
+BraveRewardsGetPublisherDonationAmountsFunction::Run() {
+  std::unique_ptr<brave_rewards::GetPublisherDonationAmounts::Params> params(
+    brave_rewards::GetPublisherDonationAmounts::Params::Create(*args_));
+
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  RewardsService* rewards_service =
+    RewardsServiceFactory::GetForProfile(profile);
+
+  if (!rewards_service) {
+    return RespondNow(Error("Rewards service is not initialized"));
+  }
+
+  rewards_service->GetPublisherBanner(params->publisher_key);
```

Comment:
I know this was not introduced in this PR, but would be awesome if you could add callback to `GetPublisherBanner` and we can remove observer for it. You would need to change `brave_donate_ui` `GetPublisherBanner` as well.

Reply:
@NejcZdovc tbh as I was working this out (and it was a bit tricky) I stopped myself to say I think we just need to let this PR move on. The methods we are using to pass information from `rewards_service_impl` -> `ledger_impl` is changing constantly and is something we should address in separate tickets. Recommend we get this PR wrapped up.